### PR TITLE
Fix/col 4 breaking change

### PR DESCRIPTION
**Pull request by maruf0011**
1. cosmos sdk drops the success params from tx data, so for now fcd will use tx code to check if a tx is success full or not
2. Implement timeout for promise operation (mostly for DB operation promises to stop long time open connection). 

NB: No.2 is implemented only on `general info collector`, `validator collector` and `price collector`. Will implement widely 
after testing the impact from tequila testnet.

### CODE REVIEW

**Comment by maruf0011** [comment]
Now promise timeout is implemented in the root process using semaphore.
Timeout lists
1. Dashboard collector 20 mins as total user query took 3 mins to calculate on each day
2. Block collector 10 mins as it contains a loop to fill up the missing block collectively
3. Unvested and richlist collector 10 mins both runs once a day and richlist has huge data to update.
All remaining process has a default 1 min timeout.

File: src/collector/network.ts [hunk 1]
```diff
@@ -17,14 +18,16 @@ async function getVolumeFromSend(timestamp: number): Promise<{ [denom: string]:
 
   const volumeByCurrency = {}
 
-  txs.forEach((tx) => {
-    const msgs = get(tx, 'data.tx.value.msg')
-    const logs = get(tx, 'data.logs')
+  txs.forEach((tx: TxEntity) => {
+    const lcdTx = get(tx, 'data') as Transaction.LcdTransaction
```
Comment (hanjukim) [inline src/collector/network.ts:22]: we dont need to use get here

File: src/collector/network.ts [hunk 2]
```diff
@@ -17,14 +18,16 @@ async function getVolumeFromSend(timestamp: number): Promise<{ [denom: string]:
...
+    const msgs = get(lcdTx, 'tx.value.msg')
+    const logs = get(lcdTx, 'logs')
```
Comment (hanjukim) [inline src/collector/network.ts:24]: this too

File: src/collector/swap.ts [hunk 3]
```diff
@@ -74,15 +75,16 @@ type SwapValueDetails = {
   fee?: DenomMap
 }
 
-function getSwapValues(tx): SwapValueDetails {
-  const logs = get(tx, 'data.logs')
-  const msgs = get(tx, 'data.tx.value.msg')
+function getSwapValues(tx: TxEntity): SwapValueDetails {
+  const lcdTx = get(tx, 'data') as Transaction.LcdTransaction
+  const logs = get(lcdTx, 'logs')
```
Comment (hanjukim) [inline src/collector/swap.ts:80]: same

File: src/service/staking/getDelegationTxs.ts [hunk 4]
```diff
@@ -76,20 +78,12 @@ function eventMapper(height: string, valOpAddr: string, timestamp) {
...
-
 export default async function getDelegationTxs(data: GetDelegationEventsParam): Promise<DelegationTxsReturn> {
   const rawTxs = await getRawDelegationTxs(data)
 
   const result = rawTxs.txs
-    .filter((tx: any) => {
-      const { success } = isSuccessful(tx.logs)
-      return success
+    .filter((tx: Transaction.LcdTransaction) => {
+      return isSuccessfulTx(tx)
```
Comment (hanjukim) [inline src/service/staking/getDelegationTxs.ts:86]: we can just apply isSuccesfulTx to filter

File: src/types/tx.d.ts [hunk 5]
```diff
@@ -45,14 +44,16 @@ declare namespace Transaction {
     height: string
     txhash: string
     raw_log: string
-    logs: Log[]
+    logs: Log[] // doesn't exist if tx failed
     gas_wanted: string
     gas_used: string
+    codespace: string
+    code: number
```
Comment (hanjukim) [inline src/types/tx.d.ts:51]: code?: number ?

**Review by hanjukim — CHANGES_REQUESTED** [review]

**Review by hanjukim — APPROVED** [review]

**Review by hanjukim — APPROVED** [review]

**Pull request merged**